CIScripts/Test/TestConfigurationUtils.ps1

```diff
@@ -215,13 +215,15 @@ function Disable-AgentService {
 
 function Get-AgentServiceStatus {
     Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
-    $Service = Invoke-Command -Session $Session -ScriptBlock {
-        Get-Service "ContrailAgent" -ErrorAction SilentlyContinue
-    }
-    if ($Service -and $Service.Status) {
-        return $Service.Status.ToString()
-    } else {
-        return $null
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        $Service = Get-Service "ContrailAgent" -ErrorAction SilentlyContinue
+
+        if ($Service -and $Service.Status) {
+            return $Service.Status.ToString()
+        } else {
+            return $null
+        }
```

Comment:
> Don't pass the Service object across the session boundary

if we do this fix, do we actually need changes to MsiExec?

Reply:
I don't know actually. I'd say a risk of Service handle not being GC-ed exists also if we don't pass it through the boundary.

Reply:
From my very imprecise testing it seems that not passing Service across the session boundary indeed lowers the average time of the test, but running GC lowers it even more.

Reply:
OK let's keep it then I guess. Nice find though